src/main/java/com/laborete/LaboreteAPI/profile/services/UsersServiceImpl.java

```diff
@@ -102,38 +129,127 @@ public ResponseEntity<HttpStatus> uploadUserAvatar(String name, MultipartFile fi
             byte[] bytes = file.getBytes();
 
             UUID uuid = UUID.randomUUID();
-            var path = this.createDirectory(uuid);
+            var path = createDirectory(uuid, ROOT_PATH_AVATARS);
+            MimetypesFileTypeMap fileTypeMap = new MimetypesFileTypeMap();
 
             File avatar = new File(path, uuid + FileUtils.getFileExtension(file));
             BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(avatar));
 
             stream.write(bytes);
             stream.close();
 
-            UserAvatarEntity userAvatarEntity = new UserAvatarEntity(uuid, name, file.getSize());
+
+            UserAvatarEntity userAvatarEntity = new UserAvatarEntity(
+                    uuid,
+                    name,
+                    file.getSize(),
+                    FileUtils.getFileExtension(file),
+                    fileTypeMap.getContentType(avatar));
             user.setUserAvatarEntity(userAvatarEntity);
             this.userAvatarRepository.save(userAvatarEntity);
             this.usersRepository.save(user);
+            usersRepository.flush();
+
 
             return new ResponseEntity<>(HttpStatus.CREATED);
         } catch (Exception e) {
             throw new ResourceFileUploadErrorException(FILE_FAILED_UPLOAD, e);
         }
     }
 
-    private String createDirectory(UUID uuid) {
+    @Transactional
+    public ResponseEntity<HttpStatus> uploadUserBackground(String name, MultipartFile file, UUID userId) {
+        UserEntity user = this.usersRepository.getUserById(userId)
+                .orElseThrow(
+                        () -> new ResourceNotFoundException(USER_NOT_FOUND + userId)
```

Comment:
SonarQube would register a minor code smell because here we concat strings via _+_ operator. Please find an alternative way (_String.join_, _String.format_ or something similar).